mirdata/datasets/tonas.py

```diff
@@ -0,0 +1,409 @@
+    author       = {COFLA (COmputational analysis of FLAmenco music) team},
+    title        = {{TONAS: a dataset of flamenco a cappella sung 
+                     melodies with corresponding manual transcriptions}},
+    month        = mar,
+    year         = 2013,
+    publisher    = {Zenodo},
+    version      = {1.0},
+    doi          = {10.5281/zenodo.1290722},
+    url          = {https://doi.org/10.5281/zenodo.1290722}
+}
+@inproceedings{inproceedings,
+    author = {Mora, Joaquin and Gómez, Francisco and Gómez, Emilia
+              and Borrego, Francisco Javier and Díaz-Báñez, José},
+    year = {2010},
+    month = {01},
+    pages = {351-356},
+    title = {Characterization and Similarity in A Cappella Flamenco Cantes.}
+}
+@ARTICLE{6791736,
+    author = {E. {Gómez} and J. {Bonada}},
+    journal = {Computer Music Journal},
+    title = {Towards Computer-Assisted Flamenco Transcription: An Experimental 
+           Comparison of Automatic Transcription Algorithms as Applied to A 
+           Cappella Singing},
+    year = {2013},
+    volume = {37},
+    number = {2},
+    pages = {73-90},
+    doi = {10.1162/COMJ_a_00180}}
+"""
+
+
+REMOTES = None
+
+DOWNLOAD_INFO = """
+        Unfortunately, the TONAS dataset is not available to be shared openly. However,
+        you can request access to the dataset in the following link, providing a brief
+        explanation of what your are going to use the dataset for:
+        ==> https://zenodo.org/record/1290722
+        Then, unzip the dataset and locate it to {}. If you unzip it into a different path,
+        please remember to set the right data_home when initializing the dataset.
+"""
+
+LICENSE_INFO = """
+The TONAS dataset is offered free of charge for internal non-commercial use only. You can not redistribute it nor 
+modify it. Dataset by COFLA team. Copyright © 2012 COFLA project, Universidad de Sevilla. Distribution rights granted 
+to Music Technology Group, Universitat Pompeu Fabra. All Rights Reserved.
+"""
+
+# NoteData extension for Tonas to support energy annotation
+class NoteData(annotations.NoteData):
+    def __init__(self, intervals, notes, energies, confidence=None):
+        super().__init__(intervals, notes, confidence)
+
+        annotations.validate_array_like(intervals, np.ndarray, float)
+        annotations.validate_array_like(notes, np.ndarray, float)
+        annotations.validate_array_like(energies, np.ndarray, float)
+        annotations.validate_array_like(
+            confidence, np.ndarray, float, none_allowed=True
+        )
+        annotations.validate_lengths_equal([intervals, notes, energies, confidence])
+        annotations.validate_intervals(intervals)
+        annotations.validate_confidence(confidence)
+
+        self.intervals = intervals
+        self.notes = notes
+        self.energies = energies
+        self.confidence = confidence
+
+
+# F0 extension for Tonas to support energy annotation
+class F0Data(annotations.F0Data):
```

Comment:
Especially in this one that you have the automatically estimated frequencies?

Reply:
Yes! I have updates de docstring for both extended classes, and I have specified this matter. F0 annotations have the automatically extracted F0 and then the semi-automatic F0 as well.